cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -45,22 +45,25 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
 	}
 	status, err := c.cStorPoolEventHandler(operation, cStorPoolGot)
 	if status == "" {
+		glog.Warning("Empty status recieved for csp status in sync handler")
 		return nil
 	}
-	cStorPoolGot.Status.Phase = apis.CStorPoolPhase(status)
 	if err != nil {
-		glog.Errorf(err.Error())
-		_, err := c.clientset.OpenebsV1alpha1().CStorPools().Update(cStorPoolGot)
-		if err != nil {
-			return err
-		}
-		glog.Infof("cStorPool:%v, %v; Status: %v", cStorPoolGot.Name,
-			string(cStorPoolGot.GetUID()), cStorPoolGot.Status.Phase)
 		return err
 	}
+	cStorPoolGot.Status.Phase = apis.CStorPoolPhase(status)
```

Comment:
Add the above code as in case of pool creation failure , status won't be updated on CSP.